frontend/src/metabase/dashboard/components/DashCard/DashCardMenu/DashCardMenu.tsx

```diff
@@ -129,32 +130,31 @@ interface QueryDownloadWidgetOpts {
   question: Question;
   result?: Dataset;
   isXray?: boolean;
-  isEmbed: boolean;
   /** If public sharing or static/public embed */
   isPublicOrEmbedded?: boolean;
   isEditing: boolean;
+  downloadsEnabled: boolean;
 }
 
 DashCardMenu.shouldRender = ({
   question,
   result,
   isXray,
-  isEmbed,
   isPublicOrEmbedded,
   isEditing,
+  downloadsEnabled,
 }: QueryDownloadWidgetOpts) => {
   // Do not remove this check until we completely remove the old code related to Audit V1!
   // MLv2 doesn't handle `internal` queries used for Audit V1.
   const isInternalQuery = InternalQuery.isDatasetQueryType(
     question.datasetQuery(),
   );
 
-  if (isEmbed) {
-    return isEmbed;
+  if (isPublicOrEmbedded) {
+    return downloadsEnabled;
```

Comment:
This was changed compared to the original PR.
I had initially changed the DashcardMenu to return null if the menu items had length of zero, but after some changes on master that logic didn't work anymore so I had to update the shouldRender logic